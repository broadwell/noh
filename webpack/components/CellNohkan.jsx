import React from "react";
import PropTypes from "prop-types";

const CellNohkan = props => (
  <div className={`cell cell--${props.length} cell--nohkan`}>
    <span data-tooltip={props.text} className="truncate">
      {props.text}
    </span>
  </div>
);

CellNohkan.propTypes = {
  text: PropTypes.string.isRequired,
  length: PropTypes.number.isRequired
};

export default CellNohkan;
