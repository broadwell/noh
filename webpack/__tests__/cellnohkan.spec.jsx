import React from "react";
import { shallow } from "enzyme";
import CellNohkan from "../components/CellNohkan";

describe("<CellNohkan>", () => {
  it("renders as expected by default", () => {
    const component = shallow(<CellNohkan text="Naka no takane" length={3} />);
    expect(component).toMatchSnapshot();
  });
});
