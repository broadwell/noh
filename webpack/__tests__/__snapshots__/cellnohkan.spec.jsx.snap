// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CellNohkan> renders as expected by default 1`] = `
<div
  className="cell cell--3 cell--nohkan cell--tooltip"
>
  <span
    className="truncate"
    data-tooltip="Naka no takane"
  >
    Naka no takane
  </span>
</div>
`;
